### DISCUSSION TITLE
Regarding reading python library functions

### DISCUSSION THREAD

**Discussion by sunilkumar105** [pymodbus-dev/pymodbus#1289] [Help] [answered]
Hii,
Hope this text finds you well :)
I am new to python, now every general functions are well understandable for me but still there are somepoints where i am stucked like in this python library like where are the all functions defined which can be used in program
where is .read_register() function defined , where is .write register, .read holding register etc.

where are these functions defined and in which specififc files

how to read what arguments they need

Please help me out

**Comment by janiversen** [comment, marked as answer]
I politely suggest you read the pymodbus documentation, start with the REAME.rst which will point you to the rest of the documentation.
